### DISCUSSION TITLE
Simulate array[*] with sqljson

### DISCUSSION THREAD

**Discussion by jiaxuan** [ent/ent#3237] [Q&A] [answered]
I'm trying to construct a query similar to `jsonb_path_exists(json_value, '$.attr1.array1[*].attr2 ? (@ == "abc")')`.  However `sqljson` rejects the path with `[*]` in it and requires a specific index to be specified. The examples all use `[0]`, `[1]` with arrays. Is there a way to match the entire array on any index? My data contains nested arrays at different levels.

**Comment by jiaxuan** [comment, marked as answer]
I solved it with a custom predicate.

**Comment by joshhe** [comment]
@jiaxuan I am running into the same issue. Do you mind sharing your code here with us? Much appreciated!

**Reply by jiaxuan** [reply]
```
.Query().Where(
  func(sel *sql.Selector) {
    sel.Where(
      sql.ExprP(fmt.Sprintf(`jsonb_path_exists(json_value, '$.attr1.array1[*].attr2 ? (@ == "%s")')`, expected)),
    )
  },
)
```

**Reply by joshhe** [reply]
Thanks! @a8m is there plan to support wild card array index in the future?
